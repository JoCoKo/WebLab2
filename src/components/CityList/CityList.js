import React from 'react';
import './CityList.css'
import SecondaryCity from "../SecondaryCity/SecondaryCity";

const CityList = ({cities, onDeleted}) => (
    <div className="CityList">
        {cities.map((city, index)=>(<SecondaryCity weather={city} key={`city-${index}`} onDeleted={onDeleted} i={index}/>))}
    </div>
);
export default CityList;