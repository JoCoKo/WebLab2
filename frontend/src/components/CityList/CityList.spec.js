import React from 'react';
import CityList from "./CityList";
import {shallowToJson} from "enzyme-to-json";
import {shallow} from "enzyme";

describe('CityList', () => {
  describe('с пустым массивом городов', () => {
    const props = {
      cities: [],
      onDeleted: () => {}
    };
    const cityList = shallow(<CityList {...props} />);
    test('отображается корректно', () => {
      expect(shallowToJson(cityList)).toMatchSnapshot();
    });
  });
  describe('с заполненым массивом городов', () => {
    const city = JSON.parse(
      '{"id":479561,"city":"Ufa","temp":"-6","image":"http://openweathermap.org/img/wn/04n@2x.png","wind":3,"cloudiness":"overcast clouds","pressure":1026,"humidity":85,"coordinates":"54.73,55.95"}'
      );
    const props = {
      cities: [city, city, city],
      onDeleted: () => {}
    };
    const cityList = shallow(<CityList {...props} />);
    test('отображается корректно', () => {
      expect(shallowToJson(cityList)).toMatchSnapshot();
    });
  });
});