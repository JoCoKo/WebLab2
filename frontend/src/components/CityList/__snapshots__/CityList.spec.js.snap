// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CityList с заполненым массивом городов отображается корректно 1`] = `
<div
  className="CityList"
>
  <SecondaryCity
    i={0}
    key="city-0"
    onDeleted={[Function]}
    weather={
      Object {
        "city": "Ufa",
        "cloudiness": "overcast clouds",
        "coordinates": "54.73,55.95",
        "humidity": 85,
        "id": 479561,
        "image": "http://openweathermap.org/img/wn/04n@2x.png",
        "pressure": 1026,
        "temp": "-6",
        "wind": 3,
      }
    }
  />
  <SecondaryCity
    i={1}
    key="city-1"
    onDeleted={[Function]}
    weather={
      Object {
        "city": "Ufa",
        "cloudiness": "overcast clouds",
        "coordinates": "54.73,55.95",
        "humidity": 85,
        "id": 479561,
        "image": "http://openweathermap.org/img/wn/04n@2x.png",
        "pressure": 1026,
        "temp": "-6",
        "wind": 3,
      }
    }
  />
  <SecondaryCity
    i={2}
    key="city-2"
    onDeleted={[Function]}
    weather={
      Object {
        "city": "Ufa",
        "cloudiness": "overcast clouds",
        "coordinates": "54.73,55.95",
        "humidity": 85,
        "id": 479561,
        "image": "http://openweathermap.org/img/wn/04n@2x.png",
        "pressure": 1026,
        "temp": "-6",
        "wind": 3,
      }
    }
  />
</div>
`;

exports[`CityList с пустым массивом городов отображается корректно 1`] = `
<div
  className="CityList"
/>
`;
